charts/redpanda/templates/_configmap.tpl

```diff
@@ -90,6 +90,36 @@ bootstrap.yaml: |
   {{- if and (not (hasKey .Values.config.cluster "storage_min_free_bytes")) ((include "redpanda-atleast-22-2-0" . | fromJson).bool) }}
   storage_min_free_bytes: {{ include "storage-min-free-bytes" . }}
   {{- end }}
+{{/* AUDIT LOGS */}}
+{{- if (include "redpanda-atleast-23-3-0" . | fromJson).bool }}
+  {{- if and ( dig "enabled" "false" .Values.auditLogging ) (include "sasl-enabled" $root | fromJson).bool }}
+  audit_enabled: true
+  audit_client_max_buffer_size: {{ .Values.auditLogging.clientMaxBufferSize | default 16777216 }}
+  audit_queue_drain_interval_ms: {{ .Values.auditLogging.queueDrainIntervalMs | default 500 }}
+  audit_queue_max_buffer_size_per_shard: {{ .Values.auditLogging.maxBufferSizePerShard | default 1048576 }}
+  audit_log_num_partitions: {{ .Values.auditLogging.partitions | default 12 }}
```

Comment:
I do not necessarily agree with this. How do we allow dials for customers for these situations, should i just ignore them if the defaults are set?

Reply:
I don't understand what `dials for customers` mean. As John mentioned it should be not set at all. Redpanda is responsible to handle any configuration. In extreme case only `audit_enabled: true` and `audit_log_client` should be set. With latest changes provided it is hard to maintain with every line if statement. Can we create more convenient for us configuration design were range loop could go through any additional audit configuration option?

```
auditLogging:
  enabled: true
  configuration:
    client_max_buffer_size: 888
    queue_drain_interval_ms: 777
    queue_max_buffer_size_per_shard: 666
  listeners: default
```

In the above design we would need to prefix key with `audit_`. The simple range will be open for extension when Redpanda would add new configuration option for audit logs.

Reply:
My vote would be to allow a user access to these configs.  How full these audit buffers get is very dependent on how the customer is using Redpanda.  The audit system tries its best to be conservative when it comes to the usage of the buffers (grouping the 'same' audit events together using a count field).  However, distinct audit events will still generate new messages and use up more memory reserved for these buffers.

This is also dependent on which events the user wants to audit (`audit_enabled_event_types`).  Enabling more of these event types will generate more audit events.  The defaults we selected are reasonable, but if there is a lot of traffic to different topics from different users, the audit buffers may fill up and this will cause Redpanda to start rejecting commands (either through broker_not_available errors on Kafka API or 500 errors on admin API).

Our suggestion for customers is to monitor the metrics and, if necessary, increase the size of the buffers.

Reply:
No, i would prefer not to do this. It would cause the confusion we are seeing with the configurations we find in tiered storage, where customers ask why there are configurations in two or more places and how they would interact with each other. Instead I gated the default values when applied to the configuration to resolve the "do not set defaults" concern you bring up (which i do not agree with). Remember our job here is to set `production` defaults for customers and those are not always the default values as defined by core. Given i do not know what they are i left them as the default values for the moment.